app/controllers/items/detail.js

```diff
@@ -573,15 +569,55 @@ export default GoodcityController.extend(
       /**
        * Fetches all the assoicated packages to a box/pallet
        */
-      fetchContainedPackages() {
-        this.runTask(
-          this.get("packageService")
-            .fetchContainedPackages(this.get("item.id"))
-            .then(data => {
-              this.get("store").pushPayload(data);
-              this.set("associatedPackages", data.items);
-            })
+      fetchContainedPackages(page = 1) {
+        // Set the associatedPackages to []
+        // This is to flush out all the old records iff page is 1
+        if (page === 1) {
+          this.set("associatedPackages", []);
```

Comment:
maybe the full list that exists inside the infinite-scroll component could be made accessible to the controller for these kinds of operations. To avoid you having to work around it. Maybe check with @abulsayyad123

Reply:
you could also have your `fetchContainedPackages` function return wrapped records, that way you can add your own properties (locally) to mark them as deleted.

e.g

the function returns a list like this

```javascript
[
   {
       record: record,
       deleted: false
   }
   ...
]
```

After a successful deletion you can set `deleted` to `true`.

And in the handlebars file you can do

```
{{#unless row.deleted}}
   <div> ... </div>
{{/unless}}
```

That way you wouldn't need to store a "parallel" list to the component's

Reply:
Actually... I just checked the ember docs, and it seems records all have an `isDeleted` property.

So simply by having a handlebars `{{#unless}}` you can hide them from the UI without needing to keep track of a list

https://guides.emberjs.com/release/models/creating-updating-and-deleting-records/

Reply:
I made it to return `isDeleted` 
The package-service method doesn't return an array of EmberObject, but the response from the API, so it wont have the isDeleted

I added ur first suggestion : )